lib/linkerd2-trace-context/src/propagation.rs

```diff
@@ -215,10 +215,20 @@ fn get_header_str<'a, B>(request: &'a http::Request<B>, header: &str) -> Option<
         .ok()
 }
 
-fn parse_header_id<B>(request: &http::Request<B>, header: &str) -> Option<Id> {
+fn parse_header_id<B>(request: &http::Request<B>, header: &str, pad_to: usize) -> Option<Id> {
```

Comment:
would feel much better about this with unit tests, but not a blocker to fixing the bug.